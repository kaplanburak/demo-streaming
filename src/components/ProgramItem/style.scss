@import "../../styles/main.scss";

#programs__item {
  width: 100%;
  cursor: pointer;

  img {
    width: 100%;
    max-width: 14rem;
    @include light-shadow();
  }

  div {
    z-index: 100;
    padding-top: 0.5rem;

    span {
      margin: 0;
      font-weight: 500;
    }
  }
}

@media only screen and (max-width: 500px) {
  #programs__item {
    display: grid;
    place-items: center;
  }
}
