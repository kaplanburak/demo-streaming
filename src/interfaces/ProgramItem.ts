export interface IProgramItemProps {
  title: string;
  url: string;
}
