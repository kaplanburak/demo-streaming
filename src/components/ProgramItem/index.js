import React from "react";
import "./style.scss";

const ProgramItem = ({ title, url }) => {
  return (
    <div id="item-wrapper">
      <div id="poster">
        <img src={url} alt={title} />
      </div>
      <div id="title">
        <span>{title}</span>
      </div>
    </div>
  );
};

export default ProgramItem;
