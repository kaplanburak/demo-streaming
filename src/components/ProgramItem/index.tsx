import React from "react";
import "./style.scss";
import { IProgramItemProps } from "../../interfaces";

const ProgramItem: React.FC<IProgramItemProps> = ({ title, url }): JSX.Element => {
  return (
    <div id="poster">
      <img src={url} alt={title} />
      <div id="title">
        <span>{title}</span>
      </div>
    </div>
  );
};

export default ProgramItem;
