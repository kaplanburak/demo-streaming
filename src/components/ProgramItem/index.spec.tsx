import React from "react";
import { shallow } from "enzyme";
import ProgramItem from "./index";

describe("ProgramItem component", () => {
  const programItem = shallow(<ProgramItem title={"test"} url="https://google.com" />);

  it("should render with props", () => {
    expect(programItem).toMatchSnapshot();
  });

  it("should have the url as image source", () => {
    expect(
      programItem
        .find("#poster")
        .childAt(0)
        .props().src
    ).toEqual("https://google.com");
  });
});
