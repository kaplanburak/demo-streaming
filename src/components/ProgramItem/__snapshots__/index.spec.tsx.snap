// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramItem component should render with props 1`] = `ShallowWrapper {}`;
